### DISCUSSION TITLE
Reloading when assets change

### DISCUSSION THREAD

**Discussion by gregmsanderson** [livewire/livewire#7851] [1. Help]
Hello,

I was reading the docs and came across: https://livewire.laravel.com/docs/navigate#reloading-when-assets-change

That sounds great and is indeed what you want: the site to reload when the assets have changed.

_However_ it then says;

> Only query string changes are tracked

I mention this because I just tried installing a new Laravel app, installed the Breeze starter kit and chose Volt (to get Livewire). After running `npm run build` I noticed that the hash is in the file's _name_ by default, _not_ in the query string:

```bash
vite v5.0.12 building for production...
✓ 47 modules transformed.
public/build/manifest.json             0.26 kB │ gzip:  0.14 kB
public/build/assets/app-UmgUTBo1.css  35.19 kB │ gzip:  6.41 kB
public/build/assets/app-gkggixxs.js   29.83 kB │ gzip: 12.03 kB
```

Sure enough ...

```html
@vite(['resources/css/app.css', 'resources/js/app.js'])
```

... generates e.g ...

```html
<script type="module" src="https://example.test/build/assets/app-gkggixxs.js" data-navigate-track="reload"></script>
```

I found the same _without_ Breeze. The default Laravel Vite config produces the same names for the .js and .css.

So ... are the docs correct and the assets would _not_ be reloaded by Livewire? If so ... should a change in the file's name be supported too (as that is the default Laravel setup)?

Since I'd assume it would work "out of the box", especially since the `data-navigate-track="reload` _is_ present.

Thanks.

**Comment by joshhanley** [comment]
@gregmsanderson I would imagine it would work fine, as a browser sees a file with a different file name as a new file, so it should load it fine.

**Comment by gregmsanderson** [comment]
Thanks @joshhanley for the update.

I probably should have clarified that this would be using _wire:navigate_ ... where the browser _won't_ do a full page load (as I understand it). I assume that is why Livewire has this feature (since a browser would also treat a different _query_ string as different).

**Reply by joshhanley** [reply]
@gregmsanderson yep I was talking about with `wire:navigate` 🙂 when naviate swaps the URL for the script (as it has a different file name), it should then load the script with the new name.
